### PR TITLE
BE-1913 Count fds for things in supervisor.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by kenatbasis** [comment]
i believe the standard file name is `celeryfdcountcollector.py`

also, have you tested this? I've been copying their unit tests and modifying them accordingly, but instead of running unit tests it was more convenient for me to actually deploy since I was already testing the deployed version.

**Comment by ghost** [comment]
I have run the function independently, but not as a collector.

**Comment by ghost** [comment]
Also didn't I name it `celeryfdcountcollector.py`?

File: src/collectors/supervisorfdcount/supervisorfdcountcollector.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        Overrides the Collector.collect method
+        """
+        try:
+            name_pids = re.findall(
+                r'^(?P<name>\S+)\s+RUNNING\s+pid\s+(?P<pid>\d+)',
+                subprocess.check_output(['sudo', 'supervisorctl', 'status']),
+                flags=re.MULTILINE)
+            for name, pid in name_pids:
+                fd_count = len(os.listdir(os.path.join('/proc', pid, 'fd')))
+                self.publish('%s count' % name, fd_count)
```
Comment (ghost) [inline src/collectors/supervisorfdcount/supervisorfdcountcollector.py]: Is it okay that there's a space here? `%s count`

**Comment by ghost** [comment]
I've decided to change it to just catch everything in supervisor, seems like a better idea for us. Will host also get included automatically here?

**Comment by kenatbasis** [comment]
derp i meant `celeryfdcount.py`.

File: src/collectors/supervisorfdcount/supervisorfdcountcollector.py (hunk 1 above)
Comment (kenatbasis) [inline src/collectors/supervisorfdcount/supervisorfdcountcollector.py]: it is not ok, it needs to be a period delimited string

**Comment by kenatbasis** [comment]
host is automatically included

File: src/collectors/supervisorfdcount/supervisorfdcountcollector.py (hunk 1 above)
Comment (ghost) [inline src/collectors/supervisorfdcount/supervisorfdcountcollector.py]: Does it matter that this can say e.g. `celery:celery-0`?

**Comment by kenatbasis** [comment]
unfortunately, diamond run as the diamond user and does not have sudo access, so it won't be able to see supervisorctl

**Pull request closed without merging**